apm-agent-core/src/test/java/co/elastic/apm/agent/bci/methodmatching/MethodMatcherInstrumentationTest.java

```diff
@@ -0,0 +1,82 @@
+/*-
+ * #%L
+ * Elastic APM Java agent
+ * %%
+ * Copyright (C) 2018 Elastic and contributors
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package co.elastic.apm.agent.bci.methodmatching;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Method;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class MethodMatcherInstrumentationTest {
```

Comment:
@eyalkoren This tests modifiers and arguments so I don't think we need to do this again in `TraceMethodInstrumentationTest`

Reply:
Not really- it tests the matcher code, not the instrumentation code.

Reply:
Wouldn't testing it in `TraceMethodInstrumentationTest` also just test the matcher? Anyway, it probably makes sense to test one simple case for the modifier and the arguments in `TraceMethodInstrumentationTest` as well.

Reply:
It is not testing the code at `TraceMethodInstrumentation.getMethodMatcher`

Reply:
actually it's doing exactly that :)

https://github.com/elastic/apm-agent-java/pull/398/files/e4293fca3fdff5371f8c24520e8f3ed4b26c190c#diff-340c555184d8a9790f0bc3f0e2f4020cR71

Reply:
Your'e right, missed that 😊